ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_curvature_bases.m

```diff
@@ -3,14 +3,24 @@
     % Specify orientation as midpoint-tangent unless specified otherwise
     if ~exist('orientation','var')
         orientation = 'midpoint-tangent';
+    elseif or(isequal(orientation,'com-mean'), isequal(orientation,'midpoint-tangent'))
+        % do nothing; orientation is a valid option
+    else
```

Comment:
I don't quite follow your design intent with the logic here -- I would think that pulling from the sys file would be an explicit option, and not coded as a default catching term.

Reply:
I was trying to avoid changing the how the existing function parameters of illustrate_backbone_shapespace were set up. I used the existing orientation parameter with catching like this because a named orientation like midpoint-tangent can't be used at the same time as the calculated one.

I'm still not sure how function inputs are supposed to work (gracefully) in matlab. Would something like [an input parser](https://www.mathworks.com/help/matlab/ref/inputparser.html) be a good option here?

Reply:
The ~exist('orientation','var') condition and following code is acting to set the default value for the "orientation argument" if it is not set by the caller (because it is valid matlab syntax to call a function with trailing arguments not specified), and so shouldn't need to be extended from my original code.

I see why you've got the from-sys as a default option -- it allows us to put an arbitrary string in and have it be treated as a system file unless otherwise specified. I would implement this by moving your code from the "else" condition and putting it into an "otherwise" case at the end of the "switch orientation" code block (combining it with code from the "from-sys" case you already have)